pkg/sbom/cyclonedx/marshal.go

```diff
@@ -144,12 +144,20 @@ func (e *Marshaler) marshalPackages(metadata types.Metadata, result types.Result
 
 	// Group vulnerabilities by package ID
 	vulns := lo.GroupBy(result.Vulnerabilities, func(v types.DetectedVulnerability) string {
-		return lo.Ternary(v.PkgID == "", fmt.Sprintf("%s@%s", v.PkgName, v.InstalledVersion), v.PkgID)
+		pkgID := lo.Ternary(v.PkgID == "", fmt.Sprintf("%s@%s", v.PkgName, v.InstalledVersion), v.PkgID)
+		if v.PkgPath != "" {
+			pkgID = fmt.Sprintf("%s@%s", pkgID, v.PkgPath)
+		}
+		return pkgID
 	})
 
 	// Create package map
 	pkgs := lo.SliceToMap(result.Packages, func(pkg ftypes.Package) (string, Package) {
 		pkgID := lo.Ternary(pkg.ID == "", fmt.Sprintf("%s@%s", pkg.Name, utils.FormatVersion(pkg)), pkg.ID)
+		// To avoid skip same packages with different paths
+		if pkg.FilePath != "" {
```

Comment:
This only applies to libraries (jar, package.json, etc.) because pkg.FilePath is only saved for libraries:
https://github.com/aquasecurity/trivy/blob/abf227e06ec205c712f2434a60ca739a8ea4fd50/pkg/fanal/types/artifact.go#L91-L92

Reply:
`pkgs` map is used only for dependency iteration.
But `Libraries` don't contain dependencies (`DependsOn` fields).

So it doesn't break logic.

Reply:
I thought about saving slice of `ftypes.Package`'s or slice of `FilePath`'s, but i understood that we still can't get correct package, because DependsOn doesn't currently contain info about filepath.

Looks like this update is enough for now.

If we will add `Package` with `DependsOn` - we will need to update `DependsOn` struct (and logic for this) and we can update logic for CycloneDX.